core/src/main/kotlin/org/sert2521/sertain/control/PidfController.kt

```diff
@@ -21,6 +23,6 @@ class PidfController(config: PidfConfig, val dt: Double) {
         integral += error * dt
         val derivative = (error - lastError) / dt
         lastError = error
-        return (kp * error) + (ki * integral) + (kd * derivative) + kf
+        return (kp * error) + (ki * integral) + (kd * derivative) + (kf * sign(error))
```

Comment:
Should feedforward take the sign of the error or the setpoint?

Reply:
I think it is the error, because the feedforward's direction should depend on which direction the setpoint is in. But should the feedforward be subtracted? Because if the error is negative that means the feedforward should move in the positive direction, right?

Reply:
Actually I just realized it you're multiplying it by the sign of the error, you're essentially just modifying the P value. Feedforward should definitely be multiplied by the sign of the setpoint, because it's used as a "push" in the right direction. Any overshoot will be corrected by P.

Reply:
https://phoenix-documentation.readthedocs.io/en/latest/ch16_ClosedLoop.html

"Velocity Closed Loop: kF is multiplied by target velocity and added to output."

So technically it should be multiplied by the setpoint, not just the sign of the setpoint

Reply:
👍

Reply:
@andrewda So just so I understand this correctly, we should have `+ kf (setpoint * error.sign)`